source/common/upstream/health_checker_impl.cc

```diff
@@ -347,7 +347,9 @@ TcpHealthCheckerImpl::TcpHealthCheckerImpl(const Cluster& cluster,
 
 TcpHealthCheckerImpl::TcpActiveHealthCheckSession::~TcpActiveHealthCheckSession() {
   if (client_) {
-    client_->close(Network::ConnectionCloseType::NoFlush);
+    Network::ClientConnectionPtr clientToDestroy(std::move(client_));
```

Comment:
nit: `client_to_destroy`. Can you potentially have this in a common function with a comment on why we are doing this and then use that function in the other places we do the same thing?

Reply:
Good idea, but following the `expect_reset_` pattern in your other comment eliminated this repetition so I think this suggestion no longer applies.